File: app/src/main/java/com/ihadzhi/eatlimination/ui/StartActivity.java
package com.ihadzhi.eatlimination.ui;

import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.Toolbar;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentTransaction;
import androidx.navigation.NavController;
import androidx.navigation.Navigation;
import androidx.navigation.ui.NavigationUI;

import android.app.Activity;
import android.os.Bundle;
import android.view.View;
import android.view.inputmethod.InputMethodManager;
import android.widget.ProgressBar;

import com.google.android.material.bottomnavigation.BottomNavigationView;
import com.ihadzhi.eatlimination.R;

public class StartActivity extends AppCompatActivity implements ContainerInteractor {

    private BottomNavigationView bottomNavigationView;
    private ProgressBar progressBar;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_start);
        bottomNavigationView = findViewById(R.id.bottom_navigation);
        progressBar = findViewById(R.id.progress_bar);
        NavController navController = Navigation.findNavController(this, R.id.nav_host_fragment);
        NavigationUI.setupWithNavController(bottomNavigationView, navController);
    }

    @Override
    public void hideBottomNavigation() {
        bottomNavigationView.setVisibility(View.GONE);
    }

    @Override
    public void showBottomNavigation() {
        bottomNavigationView.setVisibility(View.VISIBLE);
    }

    @Override
    public void showLoadingIndicator() {
        progressBar.setVisibility(View.VISIBLE);
        progressBar.animate();
    }

    @Override
    public void hideLoadingIndicator() {
        progressBar.setVisibility(View.GONE);
    }

    @Override
    public void hideSoftKeyboard() {
        InputMethodManager imm = (InputMethodManager) getSystemService(Activity.INPUT_METHOD_SERVICE);
        imm.hideSoftInputFromWindow(getWindow().getDecorView().getWindowToken(), 0);
    }
}

File: app/src/main/java/com/ihadzhi/eatlimination/ui/StartActivity.kt
package com.ihadzhi.eatlimination.ui

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import com.ihadzhi.eatlimination.R
import androidx.navigation.Navigation
import androidx.navigation.ui.NavigationUI
import android.content.DialogInterface
import android.view.View
import android.view.inputmethod.InputMethodManager
import androidx.appcompat.app.AlertDialog
import kotlinx.android.synthetic.main.activity_start.*

open class StartActivity : AppCompatActivity(), ContainerInteractor {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_start)
        val navController = Navigation.findNavController(this, R.id.nav_host_fragment)
        NavigationUI.setupWithNavController(bottom_navigation, navController)
    }

    override fun hideBottomNavigation() {
        bottom_navigation.visibility = View.GONE
    }

    override fun showBottomNavigation() {
        bottom_navigation.visibility = View.VISIBLE
    }

    override fun showLoadingIndicator() {
        progress_bar.visibility = View.VISIBLE
        progress_bar.animate()
    }

    override fun hideLoadingIndicator() {
        progress_bar.visibility = View.GONE
    }

    override fun hideSoftKeyboard() {
        val imm = getSystemService(INPUT_METHOD_SERVICE) as InputMethodManager
        imm.hideSoftInputFromWindow(window.decorView.windowToken, 0)
    }

    override fun errorGeneric() {
        error(R.string.generic_error)
    }

    override fun error(errorMessage: Int) {
        val alertDialog = AlertDialog.Builder(this).create()
        alertDialog.setMessage(getString(errorMessage))
        alertDialog.setButton(DialogInterface.BUTTON_POSITIVE, getString(R.string.ok_button)) { dialog: DialogInterface?, which: Int -> alertDialog.dismiss() }
        alertDialog.setCanceledOnTouchOutside(true)
        alertDialog.show()
    }
}